rpc/backend/tx_info.go

```diff
@@ -156,7 +170,19 @@ func (b *Backend) GetTransactionReceipt(hash common.Hash) (map[string]interface{
 		b.logger.Debug("decoding failed", "error", err.Error())
 		return nil, fmt.Errorf("failed to decode tx: %w", err)
 	}
-	ethMsg := tx.GetMsgs()[res.MsgIndex].(*evmtypes.MsgEthereumTx)
+
+	ethMsg, isEth := tx.GetMsgs()[res.MsgIndex].(*evmtypes.MsgEthereumTx)
+	_, isBankSend := tx.GetMsgs()[res.MsgIndex].(*banktypes.MsgSend)
+	if !isEth && !isBankSend {
+		return nil, errors.New("invalid ethereum tx")
+	}
+
+	if isBankSend {
+		ethMsg, err = utilstx.BankSendTxAsEthereumTx(tx)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
to avoid the code duplication it could be better to introduce a helper method

```
func GetEthMsgOrConvertBankSendToEthMsg(tx sdk.Tx, msg sdk.Msg) (*evmtypes.MsgEthereumTx, error) {
    if ethMsg, isEth = msg.(*evmtypes.MsgEthereumTx, error); isEth {
        return ethMsg, nil
    }

    if _, isBankSend = msg.(*banktypes.MsgSend); isBankSend {
        return utilstx.BankSendTxAsEthereumTx
    }

    return nil, fmt.Errorf("neither ethereum tx nor bank transfer; got: %T", msg)
}
```


```suggestion

	ethMsg, err := GetEthMsgOrConvertBankSendToEthMsg(tx, tx.GetMsgs()[res.MsgIndex])
    if err != nil {
        return nil, err
    }
```